### PR TITLE
Add automated tests to confirm Role's can be `default`

### LINKED ISSUE #5291: Unable to set global role as a default role
**Setup**
- Rancher version: v2.6.4-rc4
- Browser type & version: Chrome Version 98.0.4758.109 (Official Build) (x86_64)

**Describe the bug**
When attempting to set a global role to the default for new users, the radio button is unable to be changed (either from yes to no, or no to yes).

**To Reproduce**
1. Launch rancher/rancher:v2.6.4-rc4
2. Login as an admin user
3. Open the sidebar, and select `Users & Authentication`
4. In the side menu, select `Roles`
5. In the resulting menu, choose a global role and click on the display name.
6. In the resulting screen, click the 3 dots at the upper right hand of the role and select "Edit Config".
7. In the resulting screen, attempt to change the value in the `New User Default` section (either from true to false or false to true).

**Result**

Regardless of the user's role, the Global Role's status as the New User Default cannot be changed.

**Expected Result**

The Global Role's status as the New User Default should be able to be changed (i.e. your choice should take effect and the role should be able to be altered).

**Screenshots**
Screenshot

**Additional context**
- The following error is displayed when trying to click the button:
```
TypeError: Cannot use 'in' operator to search for 'newUserDefault' in null
    at Function.Pt [as set] (vue.runtime.esm.js:1076:7)
    at value (management.cattle.io.globalrole.js:54:9)
    at Qt (vue.runtime.esm.js:1854:26)
    at f.n (vue.runtime.esm.js:2179:14)
    at Qt (vue.runtime.esm.js:1854:26)
    at f.t.$emit (vue.runtime.esm.js:3888:9)
    at f.clicked (RadioButton.vue:84:12)
    at click (RadioButton.vue?7efb:1:547)
    at Qt (vue.runtime.esm.js:1854:26)
    at HTMLLabelElement.n (vue.runtime.esm.js:2179:14)
```
- Error is not present on the old UI
- Error does not occur in 2.6.2, but does in 2.6.3 (in other words, looks like this started with 2.6.3)

**Context**
Users may wish to change the default global role to ensure that new users are allocated a common selection of permissions (by default). With this bug in place, it is difficult for users to do so.

### PR DESCRIPTION
**Pull request by richard-cox**
### Summary
Fixes #5291

### Occurred changes and/or fixed issues
- There were some existing e2e tests that create various roles
- Added checks to confirm that the setting is saved correctly and in the case of global roles is applied when creating a user

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: cypress/e2e/po/components/global-role-binding.po.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import ComponentPo from '@/cypress/e2e/po/components/component.po';
+import CheckboxInputPo from '~/cypress/e2e/po/components/checkbox-input.po';
```
Comment (aalves08) [inline cypress/e2e/po/components/global-role-binding.po.ts:2]: just nitpick but shouldn't we use `@/cypress/...`?

File: cypress/e2e/po/edit/management.cattle.io.user.po.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import LabeledInputPo from '@/cypress/e2e/po/components/labeled-input.po';
 import AsyncButtonPo from '@/cypress/e2e/po/components/async-button.po';
 import CheckboxInputPo from '@/cypress/e2e/po/components/checkbox-input.po';
 import { CypressChainable } from '@/cypress/e2e/po/po.types';
+import GlobalRoleBindings from '~/cypress/e2e/po/components/global-role-binding.po';
```
Comment (aalves08) [inline cypress/e2e/po/edit/management.cattle.io.user.po.ts:6]: just nitpick but shouldn't we use `@/cypress/...`?

File: cypress/e2e/tests/pages/users-and-auth/roles.spec.ts [hunk 3]
```diff
@@ -51,7 +52,7 @@ describe('Roles', { tags: ['@usersAndAuths', '@adminUser'] }, () => {
 
       // testing https://github.com/rancher/dashboard/issues/9800
       // confirming that the globalRole created is not flagged as built-in
-      roles.list().details(globalRoleName, 4).should('not.contain', 'i.icon-checkmark');
+      roles.list().checkBuiltIn(globalRoleName, false);
```
Comment (aalves08) [inline cypress/e2e/tests/pages/users-and-auth/roles.spec.ts:55]: Thanks for adjusting this mate 🙇 🙏 Logic and semantics are nicer

**Review by aalves08 — DISMISSED** [review]
Couple of small comments and a kudos

File: cypress/e2e/po/components/global-role-binding.po.ts (hunk 1 above)
Comment (richard-cox) [reply cypress/e2e/po/components/global-role-binding.po.ts]: Good catch! Have updated both this and below

**Pull request merged**